src/Traits/AuthTrait.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atymic\Twitter\Traits;
+
+use Atymic\Twitter\Exception\AuthException;
+use Atymic\Twitter\Twitter;
+
+trait AuthTrait
+{
+    /**
+     * Get a sign-in request token from Twitter.
+     *
+     * @param string $callbackUrl
+     *
+     * @throws AuthException
+     * @return array
+     */
+    public function getRequestToken(string $callbackUrl): array
+    {
+        $tokenEndpoint = $this->config->getRequestTokenUrl() ?? '';
+        $responseBody = $this->get(
+            $tokenEndpoint,
+            [
+                Twitter::KEY_OAUTH_CALLBACK => $callbackUrl,
+                Twitter::KEY_FORMAT => self::RESPONSE_FORMAT_JSON,
+                Twitter::KEY_PREBUILT_ENDPOINT => true,
+            ]
+        );
+
+        parse_str($responseBody, $token);
+        if (isset($token[Twitter::KEY_OAUTH_TOKEN], $token[Twitter::KEY_OAUTH_TOKEN_SECRET])) {
+            return $token;
+        }
+
+        throw new AuthException(sprintf('Failed to fetch request token. Response content: %s', $responseBody));
+    }
+
+    /**
+     * @param array $requestToken
+     *
+     * @return string
+     */
+    public function getAuthenticateUrl(array $requestToken): string
```

Comment:
Is there a reason why this takes an array instead of a string?

Reply:
Another case of replicating the old design for this PR. Improvements can be made for sure